idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtGenerateMembersHandler.kt

```diff
@@ -0,0 +1,315 @@
+                }
+            }
+            commands to insertedBlocks
+        }
+        runWriteAction {
+            commands.forEach { it.invokeShortening() }
+            val project = classOrObject.project
+            val codeStyleManager = CodeStyleManager.getInstance(project)
+            insertedBlocks.forEach { it.declarations.forEach { it.element?.let { element -> codeStyleManager.reformat(element) } } }
+            insertedBlocks.firstOrNull()?.declarations?.firstNotNullResult { it.element }?.let {
+                moveCaretIntoGeneratedElement(editor, it)
+            }
+        }
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun KtAnalysisSession.generateMembers(
+        editor: Editor,
+        currentClass: KtClassOrObject,
+        selectedElements: Collection<KtClassMember>,
+        copyDoc: Boolean
+    ): List<InsertedBlock> {
+        if (selectedElements.isEmpty()) return emptyList()
+        val selectedMemberSymbolsAndGeneratedPsi: Map<KtCallableSymbol, KtCallableDeclaration> = buildMap {
+            for (it in selectedElements) {
+                put(it.symbol, generateMember(currentClass.project, it, currentClass, copyDoc))
+            }
+        }
+
+        val classBody = currentClass.body
+        val offset = editor.caretModel.offset
+
+        // Insert members at the cursor position if the cursor is within the class body.
+        if (classBody == null || isCursorInsideClassBodyExcludingBraces(classBody, offset)) {
+            return runWriteAction {
+                listOf(InsertedBlock(insertMembersAfter(editor, currentClass, selectedMemberSymbolsAndGeneratedPsi.values)))
+            }
+        }
+
+        // Insert members at positions such that the result aligns with ordering of members in super types.
+        val orderedMembers = getMembersOrderedByRelativePositionsInSuperTypes(currentClass, selectedMemberSymbolsAndGeneratedPsi)
+        return insertMembersAccordingToPreferredOrder(orderedMembers, classBody.lBrace, editor, currentClass)
+    }
+
+    private fun isCursorInsideClassBodyExcludingBraces(classBody: KtClassBody, offset: Int): Boolean {
+        return classBody.textRange.contains(offset)
+                && classBody.lBrace?.textRange?.contains(offset) == false
+                && classBody.rBrace?.textRange?.contains(offset) == false
+    }
+
+    /**
+     * Given a class and some stub implementation of overridden members, output all the callable members in the desired order. For example,
+     * consider the following code
+     *
+     * ```
+     * interface Super {
+     *   fun a()
+     *   fun b()
+     *   fun c()
+     * }
+     *
+     * class Sub: Super {
+     *   override fun b() {}
+     * }
+     * ```
+     *
+     * Now this method is invoked with `Sub` as [currentClass] and `Super.a` and `Super.c` as [newMemberSymbolsAndGeneratedPsi]. This
+     * method outputs `[NewEntry(Sub.a), ExistingEntry(Sub.b), NewEntry(Sub.c)]`.
```

Comment:
It would be before `foo` since we first try to find members before the current member `b` in the `Super`. If we swap `superSymbol.psi?.prevSibling` and `superSymbol.psi?.nextSibling` in the implementation below, then it would insert after `foo()`. Insert before is the current FE1.0 behavior.